### PR TITLE
Spring data merged

### PR DESCRIPTION
**Pull request by fedaktaras**

### CODE REVIEW

File: src/main/java/com/example/bookstoreapp/dto/BookUpdateDto.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    @NotBlank
+    private String title;
+    @Min(value = 0)
+    private BigDecimal price;
+    private String author;
+    @Pattern(regexp = "^(978|979)[-]?\\d{10}$", message = "Invalid ISBN")
+    private String isbn;
+    private String description;
+    private String coverImage;
+}
```
Comment (AnyaRatusnyak) [inline src/main/java/com/example/bookstoreapp/dto/BookUpdateDto.java:20]: It's not clear for me , why we need that Dto ? It's the same as previous one

File: src/main/java/com/example/bookstoreapp/servive/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -12,26 +14,54 @@
...
+                .map(bookMapper::toDtoBook)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Book not found with id: " + id));
+    }
+
+    @Override
+    public BookDto updateBook(BookUpdateDto bookUpdateDto, Long id) {
+        Book existingBook = bookRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Book not found with id: " + id));
```
Comment (AnyaRatusnyak) [inline src/main/java/com/example/bookstoreapp/servive/impl/BookServiceImpl.java]: you can do this check without creating existingBook variable. If not bookRepository.findById(id) throw exception

**Review by AnyaRatusnyak — COMMENTED** [review]
Looks good))

File: src/main/java/com/example/bookstoreapp/dto/BookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Pattern;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookRequestDto {
+    @NotBlank
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstoreapp/dto/BookRequestDto.java]: adding messages would be great, but optional

File: src/main/java/com/example/bookstoreapp/dto/BookRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    @NotNull
+    private String title;
+    @NotNull
+    @Min(value = 0)
+    private BigDecimal price;
+    @NotBlank
+    @NotNull
+    private String author;
+    @NotNull
+    @Pattern(regexp = "^(978|979)[-]?\\d{10}$", message = "Invalid ISBN")
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstoreapp/dto/BookRequestDto.java]: https://docs.jboss.org/hibernate/stable/validator/api/org/hibernate/validator/constraints/ISBN.html

File: src/main/java/com/example/bookstoreapp/dto/BookUpdateDto.java (hunk 1 above)
Comment (yevheniisynytsia) [reply src/main/java/com/example/bookstoreapp/dto/BookUpdateDto.java:20]: yes, you can reuse `BookRequestDto` for update

File: src/main/java/com/example/bookstoreapp/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put(TIMESTAMP_KEY, LocalDateTime.now());
+        body.put(STATUS_KEY, HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put(ERRORS_KEY, errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstoreapp/exception/CustomGlobalExceptionHandler.java:42]: please add handling for `EntityNotFoundException` and check if you have any other exceptions thrown that need to be handled

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Please point this PR not to main but to you previous branch that is not merged yet

File: src/main/java/com/example/bookstoreapp/dto/BookUpdateDto.java (hunk 1 above)
Comment (fedaktaras) [reply src/main/java/com/example/bookstoreapp/dto/BookUpdateDto.java:20]: It was misunderstanding in comments. I was suggested to create update dto while I was using BookRequestDto I understood that one is going to be for PATCH and another for PUT. As you can see one is anotated with @NotNull while another is not. At Q&A i discussed it and I let it stay in project.

File: src/main/java/com/example/bookstoreapp/servive/impl/BookServiceImpl.java (hunk 2 above)
Comment (fedaktaras) [reply src/main/java/com/example/bookstoreapp/servive/impl/BookServiceImpl.java]: > you can do this check without creating existingBook variable. If not bookRepository.findById(id) throw exception

But at this line i copy all not null values from bookUpdateDto to existingBook; And then save book.
bookMapper.updateBookFromDto(bookUpdateDto, existingBook);

**Pull request merged**